Update for mainnet

piersy opened the pull request:
* Adds env files for mainnet
* Updates migrate.sh to handle the new mechanism to deterministically select an L1 starting block for mainnet migrations.
* Removes the now un-needed MIGRATION_BLOCK_TIME values from alfajores and baklava migration env files.
* Updates docker images to the latest releases.
* Removes continuity check from migrate.sh, it is now an integral part of the pre-migrate step of the migration tool.
* Updates alfajores output from migration, the generated config changed slightly due to changes in the op-geth and optimism codebases. Baklava config didn't change since it was generated after those changes were made.
* Removes platform flag from migration script, we now have a multiplatform migrate tool image

Closes https://github.com/celo-org/celo-blockchain-planning/issues/907

piersy committed:
Update for mainnet

Adds env files for mainnet, updates migrate.sh to hanle the new
mechanism to deterministicly select an L1 starting block and removes the
now un-needed MIGRATION_BLOCK_TIME values for alfajores and baklava
migration env files.

TODO: update the images to newly released versions

piersy committed:
Bring baklava.env in line with alfajores.env

piersy commented on mainnet.env:
```diff
@@ -0,0 +1,114 @@
+# Set to 0 if EIGENDA_PROXY_ENDPOINT is defined otherwise empty. This shouldn't be set manually.
+USE_LOCAL_EIGENDA_PROXY_IF_UNSET=${EIGENDA_PROXY_ENDPOINT:+0}
+
+EIGENDA_LOCAL_DISPERSER_RPC=disperser.eigenda.xyz:443 
+EIGENDA_LOCAL_SVC_MANAGER_ADDR=0x870679e138bcdf293b7ff14dd44b70fc97e12fc0
```
@gastonponti could you verify if these are correct for mainnet?

karlb replied:
Not sure what the official source of truth is, but https://github.com/Layr-Labs/eigenlayer-middleware/blob/a7732a90f88090e75a1bed118b8db671868dda15/README.md?plain=1#L84 shows `0x870679e138bcdf293b7ff14dd44b70fc97e12fc0`

piersy replied:
Thanks @karlb!

piersy committed:
Update images to latest releases

piersy committed:
Add better error when migration config cannot be downloaded

piersy committed:
Make the migration config download safer

Ensure we always download the config, so it's always the latest

palango commented on envs/alfajores/migration-config/migration.env:
```diff
@@ -1,2 +1 @@
 MIGRATION_BLOCK_NUMBER=26384000
-MIGRATION_BLOCK_TIME=1727339320
```
Do we need to hardcode those for the testnets?

piersy replied:
We hardcoded those for the testnets [here](https://github.com/celo-org/optimism/pull/342/files#diff-e9ab083305d85cb9fea217fc18988c2748f3918b328559125eba8db4e33e6da4R313). It meant we could get rid of one flag for the migration script.

palango replied:
Ah nice, I missed that!

piersy committed:
Use printf to print colored failure message

This is consistent with other failure messages.

piersy committed:
Follow shellcheck quoting recommendation

piersy committed:
Merge remote-tracking branch 'origin/main' into piersy/update-for-deterministic-l1-starting-block

piersy committed:
Bring mainnet config in line with testnet config

piersy commented:
Having tested migrating Alfajores it seems that some things have changed since we last migrated, we'll need to investigate  a bit to see how we can resolve this.

INFO [03-17|20:25:39.698] Finished full migration successfully!

```diff
Migration failed, output rollup.json and genesis.json do not match stored versions
diff --git a/envs/alfajores/config/genesis.json b/envs/alfajores/config/genesis.json
index f71168f..1ecc9d8 100644
--- a/envs/alfajores/config/genesis.json
+++ b/envs/alfajores/config/genesis.json
@@ -122,6 +122,7 @@
     "chainId": 44787,
     "constantinopleBlock": 0,
     "daoForkSupport": true,
+    "depositContractAddress": "0x0000000000000000000000000000000000000000",
     "ecotoneTime": 1727339320,
     "eip150Block": 0,
     "eip155Block": 0,
@@ -142,8 +143,7 @@
     "petersburgBlock": 0,
     "regolithTime": 1727339320,
     "shanghaiTime": 1727339320,
-    "terminalTotalDifficulty": 0,
-    "terminalTotalDifficultyPassed": true
+    "terminalTotalDifficulty": 0
   },
   "extraData": "0xe... abridged ...080",
   "gasUsed": "0x0",
diff --git a/envs/alfajores/config/rollup.json b/envs/alfajores/config/rollup.json
index 458c9d7..b257bfc 100644
--- a/envs/alfajores/config/rollup.json
+++ b/envs/alfajores/config/rollup.json
@@ -8,12 +8,13 @@
       "hash": "0xe96cb39b59ebe02553e47424e7f57dbfbffca905c3ff350765985289754a00a3",
       "number": 26384000
     },
-    "l2_time": 1727337408,
+    "l2_time": 1727339320,
     "system_config": {
       "batcherAddr": "0x6f9cc6bfe1a0b446a36453d150905bbd6a4784e0",
       "overhead": "0x0000000000000000000000000000000000000000000000000000000000000000",
       "scalar": "0x00000000000000000000000000000000000000000000000000000000000f4240",
-      "gasLimit": 30000000
+      "gasLimit": 30000000,
+      "eip1559Params": "0x0000000000000000"
     }
   },
   "block_time": 1,
@@ -45,3 +46,4 @@
     "eip1559DenominatorCanyon": 400
   }
 }
```

piersy committed:
Remove continuity check from migrate.sh

It's now baked into the premigrate step of the celo-migrate tool.

piersy committed:
Remove platform flag for docker images in migrate.sh

We now have a multiplatform image.

karlb commented:
How could the config change in retrospect? And how did you get your new values? https://storage.googleapis.com/cel2-rollup-files/alfajores/rollup.json still shows the old ones from you diff.

piersy commented:
> How could the config change in retrospect? And how did you get your new values? https://storage.googleapis.com/cel2-rollup-files/alfajores/rollup.json still shows the old ones from you diff.

Hi @karlb,

I didn't commit the changes yet, because I think they will break the config. It looks like `terminalTotalDifficultyPassed` was dropped from the config struct (a change made by op-geth). We modified what the value of `l2_time` is set to, it's now the time of the first L2 block rather than the time of the L1 origin of the first L2 block. And I haven't yet investigated `depositConractAddress` & `eip1559Params`.

piersy commented:
Hi @karlb

So it seems that depositContractAddress, was introduced as part of pectra see https://github.com/ethereum/go-ethereum/commit/dfd33c77923986dc9c20f99c5b0db6ddc87751e8. So it's not required to be set in an Op context, so probably fine for it to be empty here.

`eip1559Params` was introduced in holocene and is zero when holocene is not yet enabled. See https://github.com/celo-org/optimism/blob/4c007e1fb01e51ca911409134036cf096e37cf16/op-service/eth/types.go#L480

So all differences except for the `l2_time` seem benign.

karlb commented:
Excellent! So many differences for something that was supposed to be mostly immutable were irritating, so it's great to have explanations for all changes.

piersy committed:
Update alfajores generated config

Chages in the optimism codebase have slightly changed the config
genreated for alfajores.

palango commented on mainnet.env:
Seems like `BEDROCK_SEQUENCER_HTTP` is missing here

piersy replied:
That value is actually kept in the network env files under `envs/<network>` I haven't added them yet since I still need to discover that value.

piersy committed:
Fix L1_BEACON_RPC_FLAG

piersy committed:
Add mainnet nework specific env files

piersy committed:
Use fixed version of migrate tool

This should provided correct migrations for the testnets.

piersy committed:
Add special case for mainnet migration config URLs

Mainnet is referred to as celo in the config URL.

piersy committed:
Merge remote-tracking branch 'origin/main' into piersy/update-for-deterministic-l1-starting-block

piersy commented on envs/mainnet/op-geth.env:
```diff
@@ -0,0 +1,3 @@
+BEDROCK_SEQUENCER_HTTP=https://sequencer.cel2-sequencer-mainnet.celo-testnet.org
```
Is there any reason that your suggestion ends with a slash? We didn't do that for the testnets

palango replied:
Removed it

piersy committed:
Correct sequencer http for mainnet

Co-authored-by: Paul Lange <palango@users.noreply.github.com>

palango commented on mainnet.env:
```diff
@@ -0,0 +1,113 @@
+###############################################################################
+
+# L1 node that the op-node (Bedrock) will get chain data from
+OP_NODE__RPC_ENDPOINT=https://ethereum-rpc.publicnode.com
```
Should we add a comment here to mention that people should change this

piersy replied:
Do they need to change that? I thought this is fine.

palango replied:
I guess we'd recommend to use a RPC provider they trust + have an account without rate-limiting or throttling 🤔

palango replied:
Or their own node

piersy replied:
Ok, I was thinking that wold be self explanatory for node operators, but I can add it.

palango replied:
Please add it, then it's written down ;)

piersy committed:
Add docs suggesting using trustworthy RPC api endpoints
